### ISSUE TITLE
Wallet Connect data is filling local storage and crashing app as result

### ISSUE THREAD

**Issue by jakub-jarzabek** [WalletConnect/walletconnect-monorepo#2727]
**Describe the bug**
With each page refresh in application that is using wallet connect, additional data is pushed to local storage from library ( To values under keys that are starting with `wc@`. As a result, local storage gets full and app crashes when trying to save additional data.

**SDK Version (if relevant)**
 - Client: NextJs 12.3.0
 - @web3modal/ethereum: "^2.4.1",
 - @web3modal/react: "^2.4.1",
 - viem: "^0.3.31",
 - wagmi: "^1.0.5",

**To Reproduce**
1. Create next js app and wrap in in `WagmiConfig`
2. Start application and run `new Blob(Object.entries(localStorage).filter(([k,v])=>k.startsWith("wc@")).map(([k,v])=>v)).size` in browser console to measure size of data stored by wallet connect
3. Refresh page
4. Run function again to observe major increase in size of stored data

**Expected behavior**
Data from wallet connect pushed to local storage is not exceeding local storage capacity (5MB)

**Screenshots**
Error after exceeding local storage limit 
![Screenshot 2023-06-22 at 15 27 00](https://github.com/WalletConnect/walletconnect-monorepo/assets/57012225/80b0e09b-0538-44eb-863c-6ffe17fff538)

Wallet connect items size in local storage after couple of refreshes (in kb)
![Screenshot 2023-06-22 at 15 29 11](https://github.com/WalletConnect/walletconnect-monorepo/assets/57012225/5952e355-3b18-4ebb-b16c-96752c90bb76)

**Additional context**
Used wallet connect config that application is wrapped with:
```ts
const chains = [polygon, mainnet, bsc, arbitrum, avalanche, gnosis];

const { publicClient } = configureChains(chains, [
    w3mProvider({
        projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID ?? "",
    }),
]);

const safeConnector = new SafeConnector({
    chains: [mainnet, polygon],
    options: {
        allowedDomains: [/gnosis-safe.io$/, /app.safe.global$/],
        debug: true,
    },
});

const walletConnectConnectors = w3mConnectors({
    projectId: process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID ?? "",
    chains,
    version: 2,
});

const wagmiConfig = createConfig({
    connectors: [...walletConnectConnectors, safeConnector],
    autoConnect: true,
    publicClient,
});

const ethereumClient = new EthereumClient(wagmiConfig, chains);

export const WalletConnector = ({ children }: Props) => {
    return (
        <>
            <WagmiConfig config={wagmiConfig}>
               {children}
            </WagmiConfig>
            <Web3Modal
                themeVariables={{
                    "--w3m-accent-color": "#5B58E0",
                    "--w3m-background-color": "#5B58E0",
                }}
                themeMode="dark"
                projectId={
                    process.env.NEXT_PUBLIC_WALLET_CONNECT_PROJECT_ID ?? ""
                }
                ethereumClient={ethereumClient}
            />
        </>
    );
};
```

**Comment by ganchoradkov** [comment]
Hey @jakub-jarzabek, this issue has been already resolved and I guess `https://app.consola.finance` have not yet been updated to latest.

**Comment by jakub-jarzabek** [comment]
Hey @ganchoradkov 
We have set up both web3modal packages to use the latest minor versions so it should be updated to the latest available (2.4.7), and issue still occurred after building and deploying app today, we had to remove wallet connect completely from the app as leaving a browser window with app open for some time still causes app to crash due to local storage capacity exceeded. Are you sure that issue is fixed? If so in what version of mentioned packages should it work?
